### ISSUE TITLE
abstract inputs away from keys

### ISSUE THREAD

**Issue by HeladoDeBrownie** [HeladoDeBrownie/Nexus#35]
inputs should be handled largely interchangeably regardless of whether they come from a keyboard, a gamepad, or some other input device. e.g., explicitly handling WASD should be replaced with handling directional movement intents, probably as translated by some type of input manager

**Comment by HeladoDeBrownie** [comment]
intents could vary between different types of widgets. for example, moving north and returning to the last input in the console are ostensibly unrelated actions but both could be mapped to up arrow

**Comment by HeladoDeBrownie** [comment]
although obviously both keyboard and gamepad should be able to navigate the ui and thus the same intent could be represented by both keyboard and gamepad, it's also probably desireable to allow multiple keys on the same keyboard to translate to the same intent. for example, W and up arrow could both mean go north
